resources/folder.rb

```diff
@@ -0,0 +1,101 @@
+# To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+property :host,           String,   default: 'localhost'
+property :port,           Integer,  default: 3000
+property :admin_user,     String,   default: 'admin'
+property :admin_password, String,   default: 'admin'
+property :folder,         Hash,     default: {}
+
+default_action :create
+
+include GrafanaCookbook::FolderApi
+include GrafanaCookbook::DashboardApi
+
+action :create do
+  grafana_options = {
+    host: new_resource.host,
+    port: new_resource.port,
+    user: new_resource.admin_user,
+    password: new_resource.admin_password,
+  }
+  # If folder's name is not provided as variable,
+  # Let's use resource name for it
+  new_folder = {
+    title: new_resource.name,
```

Comment:
what do you think about adding a property for title and making it a `name_property`?

Reply:
i followed the convention in the rest of the resources.
Also this would not work for update as it supports renaming a folder so the name property looks up the old name and the title property sets the new one... I am open to suggestions.